pageserver/src/tenant/timeline.rs

```diff
@@ -3254,19 +3262,15 @@ impl Timeline {
         let all_paths = image_layers
             .iter()
             .map(|layer| layer.path())
+            .chain(std::iter::once(
+                self.conf.timeline_path(&self.timeline_id, &self.tenant_id),
+            ))
             .collect::<Vec<_>>();
-
-        par_fsync::par_fsync_async(&all_paths)
-            .await
-            .context("fsync of newly created layer files")?;
-
-        par_fsync::par_fsync_async(&[self.conf.timeline_path(&self.timeline_id, &self.tenant_id)])
-            .await
-            .context("fsync of timeline dir")?;
+        par_fsync::par_fsync(&all_paths).context("fsync of newly created layer files")?;
```

Comment:
Why this change? We would prefer the async `par_fsync_async` variant in async code, right?

Reply:
also this part should not be reverted because we should first fsync image layers, then the timeline directory.

Reply:
I did that in https://github.com/neondatabase/neon/pull/4441/commits/95bd6c4b77e1824c7c14b26a47609fe943e45ae3 , please sanity-check